app/src/test/kotlin/com/wire/android/ui/joinDeepLink/JoinViaCodeViewModelTest.kt

```diff
@@ -0,0 +1,200 @@
+                JoinConversationViaCodeUseCase.Result.Success.Unchanged(conversationId)
+            )
+            .arrange()
+
+        viewModel.joinConversationViaCode(code, key, domain)
+
+        viewModel.state `should be equal to` JoinViaDeepLinkDialogState.Joined(conversationId)
+    }
+
+    @Test
+    fun `given invalid code, when try to join conversation, then state is updated`() = runTest {
+        val (code, key, domain) = Triple("code", "key", "domain")
+        val (arrangement, viewModel) = Arrangement()
+            .withJoinConversationCodeError(
+                JoinConversationViaCodeUseCase.Result.Failure.Generic(NetworkFailure.NoNetworkConnection(RuntimeException("Error")))
+            )
+            .arrange()
+
+        assertEquals(
+            JoinViaDeepLinkDialogState.Idle,
+            viewModel.state
+        )
+        viewModel.joinConversationViaCode(code, key, domain)
+        advanceUntilIdle()
+
+        coVerify(exactly = 1) {
+            arrangement.joinViaCode(
+                code = code,
+                key = key,
+                domain = domain,
+                password = null
+            )
+        }
+
+        viewModel.state `should be equal to` JoinViaDeepLinkDialogState.UnknownError
+    }
+
+    @Test
+    fun `given password, when try to join conversation, then password Is passed to the use case`() = runTest {
+        val (code, key, domain) = Triple("code", "key", "domain")
+        val (arrangement, viewModel) = Arrangement()
+            .withJoinConversationCodeError(
+                JoinConversationViaCodeUseCase.Result.Failure.Generic(NetworkFailure.NoNetworkConnection(RuntimeException("Error")))
+            )
+            .withPassword("password")
+            .arrange()
+
+        viewModel.joinConversationViaCode(code, key, domain)
+        advanceUntilIdle()
+
+        coVerify(exactly = 1) {
+            arrangement.joinViaCode(
+                code = code,
+                key = key,
+                domain = domain,
+                password = "password"
+            )
+        }
+    }
+
+    @Test
+    fun `given password field is empty, when try to join conversation, then password Is not passed to the use case`() = runTest {
+        val (code, key, domain) = Triple("code", "key", "domain")
+        val (arrangement, viewModel) = Arrangement()
+            .withJoinConversationCodeError(
+                JoinConversationViaCodeUseCase.Result.Failure.Generic(NetworkFailure.NoNetworkConnection(RuntimeException("Error")))
+            )
+            .withPassword("")
+            .arrange()
+
+        viewModel.joinConversationViaCode(code, key, domain)
+        advanceUntilIdle()
+
+        coVerify(exactly = 1) {
+            arrangement.joinViaCode(
+                code = code,
+                key = key,
+                domain = domain,
+                password = null
+            )
+        }
+        viewModel.passwordOrNull() `should be equal to` null
```

Comment:
this is redundant checking, as you already verified that `joinViaCode` was called with `password = null`. And you can remove `@VisibleForTest` annotation there.